### PR TITLE
Master

### PR DESCRIPTION
**Pull request by G0URAB**
This is the stable version of the application LauNutsAPI.

### CODE REVIEW

File: src/main/java/launutsapi/LaunutsController.java [hunk 1]
```diff
@@ -1,8 +1,155 @@
...
+			lau = new Lau(notification);	
+			return Arrays.asList(lau);	
+		}
+		else 
+			return Arrays.asList(lau_service.getLauJson(query_string));	
+			
+	}
+	
+	
+	@RequestMapping("/launuts/nuts/{query_string}/ttl")
```
Comment (Aamir-Developer) [inline src/main/java/launutsapi/LaunutsController.java]: This rest api is not generating the expected result. It displays: wrong format to the correct path specified.
I have also attached image below for reference.
error

File: src/main/java/launutsapi/LaunutsController.java [hunk 2]
```diff
@@ -1,8 +1,155 @@
...
+	
+			nut = new Nuts(notification);	
+			return Arrays.asList(nut);	
+		}
+		else 
+			return Arrays.asList(nut_service.getNutsJson(query_string));	
+			
+	}
+	
+	@RequestMapping("/launuts/lau/{query_string}/json")
```
Comment (Aamir-Developer) [inline src/main/java/launutsapi/LaunutsController.java]: In this API, for every lau_code, it is generating same LAU. I have also attached image below for reference.
error2

Comment (Aamir-Developer) [inline src/main/java/launutsapi/LaunutsController.java]: Same case when we write location name. I have attached an image below for reference.
error3

**Comment by Aamir-Developer** [comment]
The following nuts are working fine:
localhost:8080/launuts/nuts/de113/json
localhost:8080/launuts/nuts/berlin/json
localhost:8080/launuts/nuts/berlin/ttl
localhost:8080/launuts/nuts/ttl
localhost:8080/launuts/nuts/json

The following LAU are working fine:
localhost:8080/launuts/lau/de_09577125/ttl
localhost:8080/launuts/lau/ellingen/ttl
localhost:8080/launuts/lau/ttl
localhost:8080/launuts/lau/json

I have also tested above mentioned queries for wrong formats

File: src/main/java/launutsapi/LaunutsController.java (hunk 2 above)
Comment (G0URAB) [reply src/main/java/launutsapi/LaunutsController.java]: Thanks for catching this. It is resolved now.

Comment (G0URAB) [reply src/main/java/launutsapi/LaunutsController.java]: Yes, It was part of the same bug. It is resolved as well. Thanks for the review.

File: src/main/java/launutsapi/LaunutsController.java (hunk 1 above)
Comment (G0URAB) [reply src/main/java/launutsapi/LaunutsController.java]: The regex for this query has been fixed. Thanks for the review.

File: src/main/java/launutsapi/LaunutsController.java (hunk 2 above)
Comment (Aamir-Developer) [reply src/main/java/launutsapi/LaunutsController.java]: Ok, It is working only for english words but not for German words. For example: words with umlaut. Please see the below picture for reference.
no_umlauts

Comment (Aamir-Developer) [reply src/main/java/launutsapi/LaunutsController.java]: It is working fine now, good.

File: src/main/java/launutsapi/LaunutsController.java (hunk 1 above)
Comment (Aamir-Developer) [reply src/main/java/launutsapi/LaunutsController.java]: It works fine now.

File: src/main/java/launutsapi/LaunutsController.java (hunk 2 above)
Comment (G0URAB) [reply src/main/java/launutsapi/LaunutsController.java]: It works fine for me !!

Did you encode your query in the Postman? In the browser, it happens automatically. 

image

**Comment by Aamir-Developer** [comment]
I have checked for all example queries for both laus and nuts. It seems to be working fine.

**Pull request merged**